crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -126,6 +127,30 @@ impl InheritedVisibility {
     }
 }
 
+/// A bucket into which we group entities for the purposes of visibility.
+///
+/// Bevy's various rendering subsystems (3D, 2D, UI, etc.) want to be able to
+/// quickly winnow the set of entities to only those that the subsystem is
+/// tasked with rendering, to avoid spending time examining irrelevant entities.
+/// At the same time, Bevy wants the [`check_visibility`] system to determine
+/// all entities' visibilities at the same time, regardless of what rendering
+/// subsystem is responsible for drawing them. Additionally, your application
+/// may want to add more types of renderable objects that Bevy determines
+/// visibility for just as it does for Bevy's built-in objects.
+///
+/// The solution to this problem is *visibility classes*. A visibility class is
+/// a type, typically the type of a component, that represents the subsystem
+/// that renders it: for example, `Mesh3d`, `Mesh2d`, and `Sprite`. The
+/// [`VisibilityClass`] component stores the visibility class or classes that
+/// the entity belongs to. (Generally, an object will belong to only one
+/// visibility class, but in rare cases it may belong to multiple.)
+///
+/// When adding a new renderable component, you'll typically want to write an
+/// add-component hook that adds the type ID of that component to the
+/// [`VisibilityClass`] array. See `custom_phase_item` for an example.
+#[derive(Clone, Component, Default, Reflect, Deref, DerefMut)]
+pub struct VisibilityClass(pub SmallVec<[TypeId; 1]>);
```

Comment:
How much more work would it be to store `ComponentId` instead of `TypeId`? The former would be more introspect-able.

Reply:
@ItsDoot I just tried it, and unfortunately it's not really possible because those IDs get fed into the render world, and component IDs are per-world.

Reply:
Do you mind leaving that as a comment? Would be nice to have for future reference.

Reply:
Ok, I added a comment. I didn't make it a doc comment because I don't think users need to care about the reasons behind the choice.

Reply:
The per-world component ids are an interesting argument for an entity-partitioning strategy, rather than true multiple worlds.